src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfig/EditorConfigStorageLocationExtensions.cs

```diff
@@ -15,12 +15,9 @@ internal static class EditorConfigStorageLocationExtensions
         public static bool TryGetOption(this IEditorConfigStorageLocation editorConfigStorageLocation, AnalyzerConfigOptions analyzerConfigOptions, Type type, out object value)
         {
             // This is a workaround until we have an API for enumeratings AnalyzerConfigOptions. See https://github.com/dotnet/roslyn/issues/41840
-            var backingField = analyzerConfigOptions.GetType().GetField("_backing", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance);
-            var backing = backingField?.GetValue(analyzerConfigOptions);
-
-            if (backing is IReadOnlyDictionary<string, string> backingDictionary)
+            if (analyzerConfigOptions is AnalyzerConfigOptionsDictionary optionsDictionary)
```

Comment:
Hmm, is this going to work right? If this is reflecting against the compiler's object, but we're linking a copy into our project, is this going to fail the "is" check because the types won't unify? note the reflection before wouldn't have cared in the slightest.

The prior reflection not throwing asserts or something if it failed to me is incredibly suspicious too since we could have silently broken this in other ways.

Reply:
Good point. It doesn't work.